src/output.h

```diff
@@ -72,16 +72,24 @@ class Output : protected Pointers {
 
   MPI_Datatype createParticleStructType();
 
+  struct JSON_Metadata {
+    bool metaflag;                                    // indicates if there is metadata
+    std::string key;                                  // JSON key
+    std::vector<std::string> values;                  // JSON value
+    int *ivec;                                        // per-atom vector, indices for 'values'
+    JSON_Metadata() : metaflag(false) {}
+  };
+
   Output(class LAMMPS *);
   ~Output() override;
   void init();
-  void setup(int memflag = 1);                             // initial output before run/min
-  void write(bigint);                                      // output for current timestep
-  void write_dump(bigint);                                 // force output of dump snapshots
-  void write_restart(bigint);                              // force output of a restart file
-  void write_molecule_json(FILE *, int, int, int *);       // output molecule JSON objects to file
-  void reset_timestep(bigint);                             // reset output which depends on timestep
-  void reset_dt();                                         // reset output which depends on timestep size
+  void setup(int memflag = 1);                        // initial output before run/min
+  void write(bigint);                                 // output for current timestep
+  void write_dump(bigint);                            // force output of dump snapshots
+  void write_restart(bigint);                         // force output of a restart file
+  void write_molecule_json(FILE *, int, int, int *, JSON_Metadata = JSON_Metadata{}); // JSON dump molecules
```

Comment:
Following my suggestion from above, this would become something like:

```suggestion
  void write_molecule_json(FILE *, int, int, int *, json_metadata *mtd = nullptr); // JSON dump molecules
```